### DISCUSSION TITLE
Why is the ant design table and underneath components rerendering everything everytime?

### DISCUSSION THREAD

**Discussion by drewgallagher** [ant-design/ant-design#43281] [General]
We have an application using the ant design table. 

We can have tons of rows and columns within it at a time (50+ rows and 24 columns)

From using the react profiler it appears that the Table is rerendering everything including the Context provider it is using. 

For every rerender of a cell it's taking ~400ms 

This is causing major application lag. 

Event after memoizing functions, expressions, and components it is still causing issues. 

Is there a way to optimize this?

![image](https://github.com/ant-design/ant-design/assets/131900008/24659636-c91b-46e2-8a69-c92350a0d8a7)

**Comment by zombieJ** [comment]
Since render may consume data by closure, Table have to do full render when parent render. You can use `shouldCellUpdate` to memo render result for cell level.

And we have plan to give a [StaticTable](https://github.com/ant-design/ant-design/discussions/41500) RFC which will ignore all the usage of side effect like closure and memo only deps on used props.

**Reply by drewgallagher** [reply]
shouldCellUpdate makes a huge difference! 

Awesome about the staticTable as well. 

Thank you!!
